src/api/v4/dataview/category.js

```diff
@@ -0,0 +1,185 @@
+}
+
+Category.prototype = Object.create(Base.prototype);
+
+/**
+ * Set the dataview operation
+ *
+ * @param  {carto.operation} operation
+ * @return {carto.dataview.Category} this
+ * @api
+ */
+Category.prototype.setOperation = function (operation) {
+  this._checkOperation(operation);
+  this._options.operation = operation;
+  if (this._internalModel) {
+    this._internalModel.set('aggregation', operation);
+  }
+  return this;
+};
+
+/**
+ * Return the current dataview operation
+ *
+ * @return {carto.operation} Current dataview operation
+ * @api
+ */
+Category.prototype.getOperation = function () {
+  return this._options.operation;
+};
+
+/**
+ * Set the dataview operationColumn
+ *
+ * @param  {string} operationColumn
+ * @return {carto.dataview.Category} this
+ * @api
+ */
+Category.prototype.setOperationColumn = function (operationColumn) {
+  this._checkOperationColumn(operationColumn);
+  this._options.operationColumn = operationColumn;
+  if (this._internalModel) {
+    this._internalModel.set('aggregation_column', operationColumn);
+  }
+  return this;
+};
+
+/**
+ * Return the current dataview operationColumn
+ *
+ * @return {string} Current dataview operationColumn
+ * @api
+ */
+Category.prototype.getOperationColumn = function () {
+  return this._options.operationColumn;
+};
+
+/**
+ * Return the resulting data
+ *
+ * @return {CategoryData}
+ * @api
+ */
+Category.prototype.getData = function () {
+  if (this._internalModel) {
+    /**
+     * @typedef {object} CategoryItem
+     * @property {boolean} group
+     * @property {string} name
+     * @property {number} value
+     * @api
+     */
+    /**
+     * @typedef {object} CategoryData
+     * @property {number} count
+     * @property {number} max
+     * @property {number} min
+     * @property {number} nulls
+     * @property {string} operation
+     * @property {CategoryItem[]} result
+     * @property {string} type - Constant 'category'
+     * @api
+     */
+    var data = this._internalModel.get('data');
+    var result = _.map(data, function (item) {
+      return {
+        group: item.agg,
+        name: item.name,
+        value: item.value
+      };
+    });
+    return {
+      count: this._internalModel.get('count'),
+      max: this._internalModel.get('max'),
+      min: this._internalModel.get('min'),
+      nulls: this._internalModel.get('nulls'),
+      operation: this._options.operation,
+      result: result,
+      type: 'category'
+    };
+  }
+  return null;
+};
+
+Category.prototype.DEFAULTS = {
+  operation: constants.operation.COUNT,
+  operationColumn: 'column'
+};
+
+Category.prototype._listenToInternalModelSpecificEvents = function () {
+  this.listenTo(this._internalModel, 'change:aggregation', this._onOperationChanged);
+  this.listenTo(this._internalModel, 'change:aggregation_column', this._onOperationColumnChanged);
+};
+
+Category.prototype._onOperationChanged = function () {
+  this.trigger('operationChanged', this._options.operation);
```

Comment:
If the change is not provoked by us, `this._internalModel.get('aggregation')` and `this._options.operation` will not be synchronised.
I think we should assign the local function the value that has changed from the internal model just to be sure.

Reply:
Same with Formula category